x-pack/plugins/cases/public/components/all_cases/use_actions.tsx

```diff
@@ -23,84 +24,80 @@ import * as i18n from './translations';
 
-interface UseBulkActionsReturnValue {
-  actions: EuiTableComputedColumnType<Case>;
-}
-
-const ActionColumnComponent: React.FC<{ theCase: Case }> = ({ theCase }) => {
+const ActionColumnComponent: React.FC<{ theCase: Case; disableActions: boolean }> = ({
+  theCase,
+  disableActions,
+}) => {
   const [isPopoverOpen, setIsPopoverOpen] = useState(false);
   const tooglePopover = useCallback(() => setIsPopoverOpen(!isPopoverOpen), [isPopoverOpen]);
   const closePopover = useCallback(() => setIsPopoverOpen(false), []);
   const refreshCases = useRefreshCases();
-  const { permissions } = useCasesContext();
-  const canDelete = permissions.delete;
-  const canUpdate = permissions.update;
 
   const deleteAction = useDeleteAction({
-    isDisabled: !canDelete,
+    isDisabled: false,
     onAction: closePopover,
     onActionSuccess: refreshCases,
   });
 
   const statusAction = useStatusAction({
-    isDisabled: !canUpdate,
+    isDisabled: false,
     onAction: closePopover,
     onActionSuccess: refreshCases,
     selectedStatus: theCase.status,
   });
 
-  const getPanels = useCallback(
-    (): EuiContextMenuPanelDescriptor[] => [
-      {
-        id: 0,
-        title: i18n.ACTIONS,
-        items: [
-          ...(canUpdate
-            ? [
-                {
-                  name: (
-                    <FormattedMessage
-                      defaultMessage="Status: {status}"
-                      id="xpack.cases.allCasesView.statusWithValue"
-                      values={{ status: <b>{statuses[theCase.status]?.label ?? '-'}</b> }}
-                    />
-                  ),
-                  panel: 1,
-                  disabled: !canUpdate,
-                  key: `case-action-status-panel-${theCase.id}`,
-                  'data-test-subj': `case-action-status-panel-${theCase.id}`,
-                },
-              ]
-            : []),
-          /**
-           * A separator is added if a) there is one item above
-           * and b) there is an item below. For this to happen the
-           * user has to have delete and update permissions
-           */
-          ...(canUpdate && canDelete
-            ? [
-                {
-                  isSeparator: true as const,
-                  key: `actions-separator-${theCase.id}`,
-                  'data-test-subj': `actions-separator-${theCase.id}`,
-                },
-              ]
-            : []),
-          ...(canDelete ? [deleteAction.getAction([theCase])] : []),
-        ],
-      },
-      ...(canUpdate
-        ? [
-            {
-              id: 1,
-              title: i18n.STATUS,
-              items: statusAction.getActions([theCase]),
-            },
-          ]
-        : []),
-    ],
-    [canDelete, canUpdate, deleteAction, statusAction, theCase]
-  );
+  const canDelete = deleteAction.canDelete;
+  const canUpdate = statusAction.canUpdateStatus;
+
+  const getPanels = useCallback((): EuiContextMenuPanelDescriptor[] => {
```

Comment:
I might be misunderstanding how `useCallback` works, but I think use callback returns a memoized function. So on each rerender the memory address of the function still stays the same but not necessarily the value it returns. That's helpful when we pass the function itself to another component. We're passing the value to another component because we're calling `getPanels`

```
panels={getPanels()}
```

I believe it will call `getPanels` on each rendering of the component and the array will get recreated each time.

I think if we were passing the function `panels={getPanels}` then we'd want a `useCallback`. I think we want `useMemo` since we ultimately want an array `EuiContextMenuPanelDescriptor[]` and not a function that returns an array right?

Reply:
oh, you are right!! My mistake 🙂!